crates/aranya-client-capi/src/api/defs.rs

```
@@ -515,27 +521,54 @@ pub fn get_device_id(client: &mut Client) -> Result<DeviceId, imp::Error> {
     Ok(DeviceId(id))
 }
 
+#[aranya_capi_core::opaque(size = 24, align = 8)]
+pub type TeamConfig = Safe<imp::TeamConfig>;
+
+#[aranya_capi_core::opaque(size = 4, align = 4)]
+pub type TeamConfigBuilder = imp::TeamConfigBuilder;
+
+/// Attempts to construct a [`TeamConfig`], returning an `Error::Bug`
+/// if there are invalid parameters.
+///
+/// @param cfg a pointer to the team config builder
+/// @param out a pointer to write the team config to
+pub fn team_config_builder_build(
+    cfg: &mut TeamConfigBuilder,
+    out: &mut MaybeUninit<TeamConfig>,
+) -> Result<(), imp::Error> {
+    Safe::init(out, cfg.build()?);
+    Ok(())
+}
+
 /// Create a new graph/team with the current device as the owner.
 ///
 /// @param client the Aranya Client [`Client`].
+/// @param cfg the Team Configuration [`TeamConfig`].
 /// @param __output the team's ID [`TeamId`].
 ///
 /// @relates AranyaClient.
-pub fn create_team(client: &mut Client) -> Result<TeamId, imp::Error> {
+#[allow(unused_variables)] // TODO(nikki): once we have fields on TeamConfig, remove this for cfg
+pub fn create_team(client: &mut Client, cfg: &TeamConfig) -> Result<TeamId, imp::Error> {
     let client = client.deref_mut();
-    let id = client.rt.block_on(client.inner.create_team())?;
+    let cfg = aranya_client::TeamConfigBuilder::new().build()?;
+    let id = client.rt.block_on(client.inner.create_team(cfg))?;
     Ok(TeamId(id))
 }
 
 /// Add a team to the local device store.
 ///
+/// NOTE: this function is unfinished and will panic if called.
```

Should we return an `Unimplemented` error for now instead of panicking?

---

That's a problem for a different PR, ideally we'd just implement it, it panics because daemon side is a todo!() macro

---

Seems risky if we don't have time to get to that before MVP.
And we can't remove the API method because we're aiming for API stability in MVP.
I'd rather return an error than panic, until there is an implementation.
Unless the implementation is trivial, then we can just implement it soon.

---

Here's an issue for implementing `add_team`:
https://github.com/aranya-project/aranya/issues/167

---

yeah, either remove it or implement it